src/org/traccar/api/resource/DeviceResource.java

```diff
@@ -122,4 +121,34 @@ public Response updateTotalDistance(DeviceTotalDistance entity) throws SQLExcept
         return Response.noContent().build();
     }
 
+    @Path("/{slave : (geofences|drivers|attributes)}")
+    @POST
+    public Response add(Map<String, Long> entity) throws SQLException {
+        Context.getPermissionsManager().checkReadonly(getUserId());
+        if (entity.size() != 2) {
+            throw new IllegalArgumentException();
+        }
+        for (String key : entity.keySet()) {
+            Context.getPermissionsManager().checkPermission(key.replace("Id", ""), getUserId(), entity.get(key));
+        }
+        Context.getDataManager().linkObject(entity, true);
+        Context.getPermissionsManager().refreshPermissions(entity);
+        return Response.noContent().build();
+    }
+
+    @Path("/{slave : (geofences|drivers|attributes)}")
```

Comment:
Lets make agreement about the naming then. I think `target` is more proper for first object (user/device/group) and for second object offer `accessory`. What do you think?